Bump composer dependencies

anvit opened the pull request.

anvit committed:
Bump composer dependencies

melaniekung reviewed (approved):
lgtm, thanks @anvit

anvit commented:
Marking this as a WIP. PHP-CS Fixer has updated defaults for `operator_linebreak`, `php_unit_data_provider_method_order` that will need to be reviewed, and a few composer packages have updated their supported PHP version to be `>8.1.2` instead of `>7.4` so this is blocked by #1942 until gh unit test related changes have been addressed.

anvit committed:
Update github syntax check workflow

Update github syntax check workflow to use verbose mode.

anvit committed:
Fix mismatched type in sfPearFrontendPlugin

Fix a variable in sfPearFrontendPlugin to use empty array instead of
empty string. This follows a change in friends of symfony's repo.

anvit committed:
Update PHP CS Fixer

- Update CS Fixer rules to use the Symfony default for their new
  operator_linebreak rule which decides if which line the operators
  should go when there are linebreaks.

- Run CS Fixer to fix issues due to the new
  php_unit_data_provider_method_order rule which puts data providers at
  the end of tests instead of the beginning.
